cg/meta/workflow/raredisease.py

```diff
@@ -22,6 +35,128 @@ def __init__(
+            return
+        self.write_sample_sheet(
+            content=sample_sheet_content,
+            file_path=self.get_sample_sheet_path(case_id=case_id),
+            header=RarediseaseSampleSheetHeaders.headers(),
+        )
+        self.write_params_file(case_id=case_id, workflow_parameters=workflow_parameters.dict())
+
+    def get_sample_sheet_content_per_sample(
+        self, case: Case = "", case_sample: CaseSample = ""
+    ) -> list[list[str]]:
+        """Get sample sheet content per sample."""
+        sample_metadata: list[FastqFileMeta] = self.gather_file_metadata_for_sample(
+            case_sample.sample
+        )
+        fastq_forward_read_paths: list[str] = self.extract_read_files(
+            metadata=sample_metadata, forward_read=True
+        )
+        fastq_reverse_read_paths: list[str] = self.extract_read_files(
+            metadata=sample_metadata, reverse_read=True
+        )
+        sample_sheet_entry = RarediseaseSampleSheetEntry(
+            name=case_sample.sample.internal_id,
+            fastq_forward_read_paths=fastq_forward_read_paths,
+            fastq_reverse_read_paths=fastq_reverse_read_paths,
+            sex=self.get_sex_code(case_sample.sample.sex),
+            phenotype=self.get_phenotype_code(case_sample.status),
+            paternal_id=CaseSample.get_paternal_id(case_sample.father),
+            maternal_id=CaseSample.get_maternal_id(case_sample.mother),
+            case_id=case.internal_id,
+        )
+        return sample_sheet_entry.reformat_sample_content
+
+    def get_sample_sheet_content(
+        self,
+        case_id: str,
+    ) -> list[list[Any]]:
+        """Return Raredisease nextflow sample sheet content for a case."""
+        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        sample_sheet_content = []
+        LOG.info("Getting sample sheet information")
+        LOG.info(f"Samples linked to case {case_id}: {len(case.links)}")
+        for link in case.links:
+            sample_sheet_content.extend(
+                self.get_sample_sheet_content_per_sample(case=case, case_sample=link)
+            )
+        return sample_sheet_content
+
+    def get_workflow_parameters(self, case_id: str) -> WorkflowParameters:
+        """Return parameters."""
+        LOG.info("Getting parameters information")
+        return WorkflowParameters(
+            sample_sheet_path=self.get_sample_sheet_path(case_id=case_id),
+            outdir=self.get_case_path(case_id=case_id),
+        )
+
+    def get_params_file_path(self, case_id: str, params_file: Path | None = None) -> Path:
+        """Return parameters file or a path where the default parameters file for a case id should be located."""
+        if params_file:
+            return Path(params_file).absolute()
+        return Path((self.get_case_path(case_id)), f"{case_id}_params_file{FileExtensions.CONFIG}")
```

Comment:
```suggestion
        if params_file:
            return params_file.absolute()
       case_path: Path =self.get_case_path(case_id)
        return Path(case_path, f"{case_id}_params_file{FileExtensions.CONFIG}")
```
the params_file is supplied as Path, so no need to convert it.

Easier to read like this. We typically try to avoid encapsulating function calls, but rather initialise the return value first.